tests/cholupdates/rank_1/downdate/test_downdate.py

```diff
@@ -0,0 +1,181 @@
+"""Tests for the implementation of symmetric rank-1 downdates to a Cholesky factor"""
+
+from typing import Any, Dict, Tuple
+
+import numpy as np
+import pytest
+import scipy.linalg
+
+import cholupdates
+
+
+def test_valid_matrix_square_root(A_prime: np.ndarray, L_prime: np.ndarray):
+    """Assert that the resulting Cholesky factor right-multiplied with its transpose
+    is (up to numerical imprecisions) equal to the updated matrix, i.e. that the
+    Cholesky factor is a valid matrix square root"""
+    assert L_prime @ L_prime.T == pytest.approx(A_prime)
```

Comment:
is there a specific reason for using `pytest.approx` over `np.testing.assert_allclose`? From the [docs](https://docs.pytest.org/en/stable/reference.html#pytest-approx) it seems it lets you use `A_prime` as the reference value for the relative tolerance, however this is also the default in `np.testing.assert_allclose` if `A_prime` is given as the second argument.

Reply:
Originally, I thought that it might be nice to use as much of the pytest API as possible.
But now I'd say that it is actually nicer to use NumPy's test functions whenever testing NumPy code